### PR TITLE
Modernize Instructor Assessment Instance page

### PR DESCRIPTION
**Pull request by mylesw27**
This PR intends to convert the controller for the instructorAssessmentInstance page to TypeScript as well as convert the EJS files to HTML template literals.

### CODE REVIEW

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts [hunk 1]
```diff
@@ -0,0 +1,705 @@
...
+                  <th class="text-center">Auto-grading points</th>
+                  <th class="text-center">Manual grading points</th>
+                  <th class="text-center">Awarded points</th>
+                  <th class="text-center" colspan="2">Percentage score</th>
+                  <th></th>
+                </tr>
+              </thead>
+              <tbody>
+                ${instance_questions.map((instance_question, i_instance_question) => {
+                  return html` ${instance_question.start_new_zone && instance_question.zone_title
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts:312]: Nit (goes along with my next suggestion; may require Prettier reformatting)

```suggestion
                  return html`
                    ${instance_question.start_new_zone && instance_question.zone_title
```

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts [hunk 2]
```diff
@@ -0,0 +1,705 @@
...
+                              <a
+                                class="btn btn-xs btn-secondary"
+                                href="${resLocals.urlPrefix}/assessment/${resLocals.assessment
+                                  .id}/manual_grading/instance_question/${instance_question.id}"
+                                >Manual grading</a
+                              >
+                            `
+                          : null}
+                      </td>
+                    </tr>`;
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts:501]: Nit (goes along with previous suggestion):

```suggestion
                    </tr>
                    `;
```

File: apps/prairielearn/src/pages/instructorAssessmentInstance/editTotalPointsForm.html.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            name="points"
+            value="${resLocals.assessment_instance.points}"
+          />
+          <span class="input-group-addon">/${resLocals.assessment_instance.max_points}</span>
+        </div>
+      </div>
+      <p>
+        <small
+          >This change will be overwritten if further questions are answered by the student.</small
+        >
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/editTotalPointsForm.html.ts]: Updated locally in [6dfa82c](https://github.com/PrairieLearn/PrairieLearn/pull/9277/commits/6dfa82c90cab62f1a0b4089fc59ee6da989382de).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/editTotalScorePercForm.html.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            name="score_perc"
+            value="${resLocals.assessment_instance.score_perc}"
+          />
+          <span class="input-group-addon">%</span>
+        </div>
+      </div>
+      <p>
+        <small
+          >This change will be overwritten if further questions are answered by the student.</small
+        >
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/editTotalScorePercForm.html.ts]: Updated locally in [6dfa82c](https://github.com/PrairieLearn/PrairieLearn/pull/9277/commits/6dfa82c90cab62f1a0b4089fc59ee6da989382de).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.ts [hunk 5]
```diff
@@ -16,59 +15,22 @@ import {
...
-  zone_best_questions: z.number().nullable(),
-  zone_has_best_questions: z.boolean(),
-  zone_has_max_points: z.boolean(),
-  zone_id: IdSchema,
-  zone_max_points: z.number().nullable(),
-  zone_title: z.string().nullable(),
-});
-
-const logCsvFilename = (locals) => {
+const makeLogCsvFilename = (locals) => {
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.ts]: Updated this to a normal function in commit [6dfa82c](https://github.com/PrairieLearn/PrairieLearn/pull/9277/commits/6dfa82c90cab62f1a0b4089fc59ee6da989382de).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.ts [hunk 6]
```diff
@@ -106,21 +68,37 @@ router.get(
...
     );
 
-    res.locals.log = await selectAssessmentInstanceLog(res.locals.assessment_instance.id, false);
+    const assessmentInstanceLog = await selectAssessmentInstanceLog(
+      res.locals.assessment_instance.id,
+      false,
+    );
 
-    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+    const fingerprintColorsArray = ['red2', 'orange2', 'green2', 'blue2', 'turquoise2', 'purple2'];
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.ts]: Moved this to `instructorAssessmentInstance.html.ts` in commit [6dfa82c](https://github.com/PrairieLearn/PrairieLearn/pull/9277/commits/6dfa82c90cab62f1a0b4089fc59ee6da989382de).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.ts [hunk 7]
```diff
@@ -106,21 +68,37 @@ router.get(
...
+
+    res.send(
+      InstructorAssessmentInstance({
+        resLocals: res.locals,
+        logCsvFilename,
+        assessment_instance_stats,
+        assessment_instance_date_formatted,
+        assessment_instance_duration,
+        instance_questions,
+        assessmentInstanceLog: assessmentInstanceLog,
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.ts]: Updated locally in commig [6dfa82c](https://github.com/PrairieLearn/PrairieLearn/pull/9277/commits/6dfa82c90cab62f1a0b4089fc59ee6da989382de).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts [hunk 8]
```diff
@@ -0,0 +1,701 @@
...
+            'tablesorter/dist/css/theme.bootstrap.min.css',
+          )}"
+          rel="stylesheet"
+        />
+        <script src="${resLocals.node_modules_asset_path(
+            'tablesorter/dist/js/jquery.tablesorter.min.js',
+          )}"></script>
+        <script src="${resLocals.node_modules_asset_path(
+            'tablesorter/dist/js/jquery.tablesorter.widgets.min.js',
+          )}"></script>
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts]: That makes sense! Added an import for this in [6dfa82c](https://github.com/PrairieLearn/PrairieLearn/pull/9277/commits/6dfa82c90cab62f1a0b4089fc59ee6da989382de).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts (hunk 1 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts]: Updated formatting locally in [6dfa82c](https://github.com/PrairieLearn/PrairieLearn/pull/9277/commits/6dfa82c90cab62f1a0b4089fc59ee6da989382de).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts (hunk 2 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts]: Updated formatting locally in [6dfa82c](https://github.com/PrairieLearn/PrairieLearn/pull/9277/commits/6dfa82c90cab62f1a0b4089fc59ee6da989382de).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/editTotalPointsForm.html.ts
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/editTotalPointsForm.html.ts:1]: Moved these "components" into the main HTML file in [6dfa82c](https://github.com/PrairieLearn/PrairieLearn/pull/9277/commits/6dfa82c90cab62f1a0b4089fc59ee6da989382de).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/editTotalPointsForm.html.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/editTotalPointsForm.html.ts]: AFAICT this wasn't changed in the commit you referenced. Maybe this change got lost when you consolidated these components into the other file? Ditto for some of the other reformat nits here.

[the remark this answers is not in the document]
Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/editTotalPointsForm.html.ts]: Ah, nevermind, I see you actually made this change in b3cf48b.

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts [hunk 9]
```diff
@@ -0,0 +1,779 @@
...
+                      </td>
+                      ${row.event_name !== 'External grading results'
+                        ? html`<td style="word-break: break-all;">
+                            ${row.data != null ? JSON.stringify(row.data) : ''}
+                          </td>`
+                        : html`
+                            <td>
+                              <a
+                                class="btn btn-primary"
+                                href="<%= urlPrefix %>/grading_job/<%= row.data.id %>"
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts:667]: You missed some EJS syntax here.

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts [hunk 10]
```diff
@@ -0,0 +1,779 @@
...
+                                >View grading job ${row.data?.id}</a
+                              >
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts:669]: Nit:

```suggestion
                                >
                                  View grading job ${row.data?.id}
                                </a>
```

**Review by nwalters512 — COMMENTED** [review]
Only one blocker here, I left a few nits that you could address as well. This is almost good to go!

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts [hunk 11]
```diff
@@ -0,0 +1,779 @@
...
+  zone_best_questions: z.number().nullable(),
+  zone_has_best_questions: z.boolean(),
+  zone_has_max_points: z.boolean(),
+  zone_id: IdSchema,
+  zone_max_points: z.number().nullable(),
+  zone_title: z.string().nullable(),
+});
+type InstanceQuestionRow = z.infer<typeof InstanceQuestionRowSchema>;
+
+const fingerprintColorsArray = ['red2', 'orange2', 'green2', 'blue2', 'turquoise2', 'purple2'];
```
[the remark this answers is not in the document]
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts]: Updated name to `FINGERPRINT_COLORS` in [4f716c1](https://github.com/PrairieLearn/PrairieLearn/pull/9277/commits/4f716c16f38f011939f082816aa9c5c086f0460a).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts (hunk 9 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts]: Removed usage of EJS in [4f716c1](https://github.com/PrairieLearn/PrairieLearn/pull/9277/commits/4f716c16f38f011939f082816aa9c5c086f0460a).

File: apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts (hunk 10 above)
Comment (mylesw27) [reply apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts]: Updated formatting in [4f716c1](https://github.com/PrairieLearn/PrairieLearn/pull/9277/commits/4f716c16f38f011939f082816aa9c5c086f0460a).

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**